### PR TITLE
feat(workspace/app): add new resource to manage image server

### PR DESCRIPTION
**Pull request by wuzhuanhong**
**What this PR does / why we need it**:

Add new resource to manage image server of the Workspace APP.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add new resource.
2. add related document and acceptance test.
```

## PR Checklist

* [x] Tests added/passed.

```
 ./scripts/coverage.sh -o workspace -f TestAccResourceAppImageServer_basic
Prepare to calculate the coverage the following command:
TF_ACC=1 go test "./huaweicloud/services/acceptance/workspace" -v -coverprofile="./huaweicloud/services/acceptance/workspace/workspace_coverage.cov" -coverpkg="./huaweicloud/services/workspace" -run TestAccResourceAppImageServer_basic -timeout 360m -parallel 10
=== RUN   TestAccResourceAppImageServer_basic
=== PAUSE TestAccResourceAppImageServer_basic
=== CONT  TestAccResourceAppImageServer_basic
--- PASS: TestAccResourceAppImageServer_basic (814.76s)
PASS
coverage: 8.4% of statements in ./huaweicloud/services/workspace
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 814.848s        coverage: 8.4% of statements in ./huaweicloud/services/workspace

```

* [x] Documentation updated.
* [x] Schema updated.
* [x] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
    image

    
  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

### CODE REVIEW

**Comment by wuzhuanhong** [comment]
./scripts/coverage.sh -o workspace -f TestAccResourceAppImageServer_basic
Prepare to calculate the coverage the following command:
TF_ACC=1 go test "./huaweicloud/services/acceptance/workspace" -v -coverprofile="./huaweicloud/services/acceptance/workspace/workspace_coverage.cov" -coverpkg="./huaweicloud/services/workspace" -run TestAccResourceAppImageServer_basic -timeout 360m -parallel 10
=== RUN   TestAccResourceAppImageServer_basic
=== PAUSE TestAccResourceAppImageServer_basic
=== CONT  TestAccResourceAppImageServer_basic
--- PASS: TestAccResourceAppImageServer_basic (787.76s)
PASS
coverage: 6.5% of statements in ./huaweicloud/services/workspace
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 787.847s        coverage: 6.5% of statements in ./huaweicloud/services/workspace

File: docs/resources/workspace_app_image_server.md [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+  The upper limit of the number of additional sessions for a single server is `10` times the number of vCPUs in the server
+  specification minus the default number of sessions in the package.
+
+* `route_policy` - (Optional, List, ForceNew) Specifies the session scheduling policy of the server associated with
+  the image server. Changing this creates a new resource.  
+  This parameter is available only wnen the `is_vdi` parameter is set to **false**.  
+  The [route_policy](#app_image_server_route_policy) structure is documented below.
+
+  -> If any metric of the server exceeds the threshold, new sessions will be rejected. The sessions will
+    be automatically scheduled to other available servers.
```
Comment (Lance52259) [inline docs/resources/workspace_app_image_server.md]:
```suggestion
  -> If any metric of the server exceeds the threshold, new sessions will be rejected. The sessions will
     be automatically scheduled to other available servers.
```

**Comment by wuzhuanhong** [comment]
export HW_ENTERPRISE_MIGRATE_PROJECT_ID_TEST=fb9ed002-5a0d-4336-a99b-470a8f6d2830
$ ./scripts/coverage.sh -o workspace -f TestAccResourceAppImageServer_basic
Prepare to calculate the coverage the following command:
TF_ACC=1 go test "./huaweicloud/services/acceptance/workspace" -v -coverprofile="./huaweicloud/services/acceptance/workspace/workspace_coverage.cov" -coverpkg="./huaweicloud/services/workspace" -run TestAccResourceAppImageServer_basic -timeout 360m -parallel 10
=== RUN   TestAccResourceAppImageServer_basic
=== PAUSE TestAccResourceAppImageServer_basic
=== CONT  TestAccResourceAppImageServer_basic
--- PASS: TestAccResourceAppImageServer_basic (788.12s)
PASS
coverage: 6.0% of statements in ./huaweicloud/services/workspace
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 788.184s        coverage: 6.0% of statements in ./huaweicloud/services/workspace

**Pull request merged**